src/System.Device.Gpio/System/Device/Pwm/Channels/UnixPwmChannel.Linux.cs

```diff
@@ -0,0 +1,181 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.IO;
+using System.Threading;
+
+namespace System.Device.Pwm
+{
+    internal class UnixPwmChannel : PwmChannel
+    {
+        private readonly int _pwmChip;
+        private readonly int _pwmChannel;
+        private double _frequency;
+        private double _dutyCyclePercentage;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="UnixPwmChannel"/> class.
+        /// </summary>
+        /// <param name="pwmChip">The PWM chip number.</param>
+        /// <param name="pwmChannel">The PWM channel number.</param>
+        /// <param name="frequency">The frequency in hertz.</param>
+        /// <param name="dutyCyclePercentage">The duty cycle percentage represented as a value between 0.0 and 1.0.</param>
+        /// <param name="polarity">The polarity the channel should start in.</param>
+        public UnixPwmChannel(
+            int pwmChip,
+            int pwmChannel,
+            double frequency,
+            double dutyCyclePercentage = 0.5,
+            PwmPolarity polarity = PwmPolarity.ActiveHigh)
+        {
+            _pwmChip = pwmChip;
+            _pwmChannel = pwmChannel;
+            ValidatePwmChannel();
+            OpenChannel();
+            Thread.Sleep(100);  // TODO: Need a better solution to delay for available file.
+
+            SetFrequency(frequency);
+            DutyCyclePercentage = dutyCyclePercentage;
+
+            // NOTE: There appears to be a lower-level bug that won't allow the
+            // polarity to be changed until a valid period has been written.
+            SetPolarity(polarity);
+        }
+
+        /// <summary>
+        /// The frequency in hertz.
+        /// </summary>
+        public override double Frequency => _frequency;
+
+        /// <summary>
+        /// The duty cycle percentage represented as a value between 0.0 and 1.0.
+        /// </summary>
+        public override double DutyCyclePercentage
+        {
+            get
+            {
+                return _dutyCyclePercentage;
+            }
+            set
+            {
+                SetDutyCyclePercentage(value);
+            }
+        }
+
+        // Helpers for file paths.
```

Comment:
These paths are called in various areas.  Is this the preferred/best approach?  In not, please provide example and will update.

Reply:
Fine, although if the file handle would be open all the time (as an optimization) it would be as good if path was hard-coded in the only place where it would be used.
The only time I tried using hardware PWM I've experienced lag whenever I tried changing speed quickly - perhaps this was an issue here. (because of that I've switched to @Ellerbach's SoftPwm implementation which worked much better perhaps I'd have used hardware if this worked faster or at least the same)